lib/filter/tests/test_filters_fop.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2018 Balabit
+ * Copyright (c) 2018 Kokan
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "filter/filter-op.h"
+#include "filter/filter-expr.h"
+#include "filter/filter-pri.h"
+#include "filter/filter-expr-parser.h"
+#include "test_filters_common.h"
+#include "cfg-lexer.h"
+#include "apphook.h"
+
+#include <criterion/criterion.h>
+#include <criterion/parameterized.h>
+
+// TODO - test_optimizer.c
```

Comment:
Do you need this todo?

Reply:
Without any better options, I just copied this function from the mentioned file from #2833, the final solution will be to implement #2842. I just wanted to make a note here to not forget the currently very ugly code duplication. I will not close this comment yet. Still thinking about it.